codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/CargoTomlGenerator.kt

```diff
@@ -63,6 +64,11 @@ class CargoTomlGenerator(
                 "edition" to "2021",
                 "license" to settings.license,
                 "repository" to settings.moduleRepository,
+                "metadata" to listOfNotNull(
+                    "smithy" to listOfNotNull(
+                        "codegen-version" to smithyCodegenVersion(),
```

Comment:
Nit: You can call `Version.fullVersion()` directly rather than wrapping it in a function.